base/iterator.jl

```diff
@@ -284,19 +284,23 @@ end
 
 done(it::Cycle, state) = state[2]
 
-# Repeat an object infinitely many times
+
+# Repeated - repeat an object infinitely many times
 
 immutable Repeated{O}
     x::O
 end
 repeated(x) = Repeated(x)
+repeated(x, n::Integer) = take(repeated(x), n)
```

Comment:
I think broadening this to Integer will just give a MethodError unless the same is done for `take`